source/app/models/url.rb

```diff
@@ -0,0 +1,30 @@
+require 'net/http'
+
+class Url < ActiveRecord::Base
+  before_create :reset_click_count
+
+  validates :long_url, presence: true, format: { with: /http(|s):\/\/.+/ }
+  validate :long_url_is_uri_valid
+  validate :long_url_is_reachable
+
+  private
+
+  def reset_click_count
+    self.click_count = 0
```

Comment:
If the migration put a default of zero in the db, ActiveRecord would respect that when you make new instances of this class.